libsignal-service/src/sender.rs

```diff
@@ -691,6 +693,51 @@ where
         Ok(())
     }
 
+    /// Send `MessageRequestResponse` synchronization message with either a recipient ACI or a GroupV2 ID
+    #[tracing::instrument(skip(self))]
+    pub async fn send_message_request_response(
+        &mut self,
+        recipient: &ServiceAddress,
+        thread_aci: Option<&ServiceAddress>,
+        group_id: Option<&[u8; 64]>,
+        action: message_request_response::Type,
+    ) -> Result<(), MessageSenderError> {
+        match (&thread_aci, &group_id) {
```

Comment:
I'm okay with this for now, but sounds like we should have a `enum ThreadIdentifier { Aci(Uuid), Group(GroupID) }` for this

Reply:
I mirrored `MessageRequestResponse`  format but there's no need to. Especially so because the user can't click accept for a group and a recipient at the same time, so the check is better eliminated. I'll make it an enum.